src/model.ts

```diff
@@ -1,4 +1,62 @@
 import WebSocket = require("ws");
+
+interface TimerBase {
+  isRunning: boolean;
+  timeLeft: number;
+}
+
+class Timer implements TimerBase {
+  isRunning: boolean;
+  #startTime: number;
+  timeLeft: number;
+  #intervalID;
+
+  constructor(allotedTime: number) {
+    this.reset(allotedTime);
+  }
+
+  #getTimeElapsedSinceLastStart() {
+    return this.#startTime === 0 ? 0 : Date.now() - this.#startTime;
+  }
+  start() {
+    if (this.isRunning) {
+      return "Timer is already running";
+    }
+    this.isRunning = true;
+    this.#startTime = Date.now();
+
+    this.#intervalID = setInterval(() => {
+      this.timeLeft = this.timeLeft - this.#getTimeElapsedSinceLastStart();
+      this.#startTime = Date.now();
+      console.log(this.timeLeft);
+      if (this.timeLeft <= 0) {
+        this.stop();
+        return "Player Timed Out";
```

Comment:
I know the comment on this from the previous review is still pending.

Just as a side note returning from this async function does not actually do anything as the caller cannot intercept this return statement unless this was wrapped in a promise

Reply:
yeah I figured that, I am currently working on this